src/Games/NexusMods.Games.StardewValley/StardewValley.cs

```diff
@@ -101,7 +92,9 @@ protected override IReadOnlyDictionary<LocationId, AbsolutePath> GetLocations(IF
         var result = new Dictionary<LocationId, AbsolutePath>()
         {
             { LocationId.Game, installation.Path },
+            { LocationId.From("Mods"), installation.Path.Combine(Constants.ModsFolder) },
```

Comment:
I think this might be a problem:
- #658

We can have two ways to indicate the same `GamePath`: 
- `{LocationId.Game}, mods/modA/manifest.json`
- `{LocationId.From("Mods")}, /modA/manifest.json`

Figuring out that they are the same file required resolving the absolute path, something we only do late in the synchronizer.

A potential solution would be to have the installer normalize the `GamePaths` to the top level one, but the relationship between Game and Mods locations needs to be expressed somewhere.

Reply:
I'm not sure I fully understand the issue. The synchronizer only looks at top level locations. Currently, the advanced installer is the only place where we look at nested locations as suggested destinations.

Reply:
So if the user were to choose a nested location in the Advanced installer as the install location, the files under the `GamePaths` of `{LocationId.From("Mods")} would never get deployed.

I guess that would be the current issue

Reply:
It does mess with preview changes but it applies just fine:

![image](https://github.com/user-attachments/assets/2e196b8e-ea64-4972-a40b-000ddc6f2c60)

![image](https://github.com/user-attachments/assets/733d809c-8111-4d33-abab-c858c67658f2)

Reply:
I don't think you had any conflicts. 
`ShouldWin` uses `GamePaths` not absolute paths for example, so it would miss the conflicts completely.
Other parts of the app, like the Preview Changes view, or the Health Checks would not be equipped to resolve a `LocationId.Mods` to be inside `LocationId.Game`.

I don't think that we want to have `LoadoutItems` pointing to nested locations, like the mods one for SDV, we want all LoadoutItems that resolve to the same `AbsolutePath` to have the same `GamePath`, otherwise we would just introduce complexity everywhere we want to use `GamePaths`, just to resolve whether two paths are actually the same.